src/osd/osd_types.h

```diff
@@ -523,7 +530,8 @@ class coll_t {
   spg_t pgid;
   uint64_t removal_seq;  // note: deprecated, not encoded
 
-  string _str;  // cached string
+  char _str_buff[spg_t::calc_name_buf_size];
+  char *_str = NULL;
```

Comment:
I think I could save the _str_buff offset in a variable and use it in .to_str() and .c_str() and some other places like .parse(), so I could avoid operator=. I'll check on Monday those two options.